### PR TITLE
Zwave database updates and minor null bug fix

### PR DESCRIPTION
**Pull request by llamahunter**
...WS-100

### CODE REVIEW

File: bundles/binding/org.openhab.binding.zwave/database/dragontech/pa100.xml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+            <Item>
+                <Value>2</Value>
+                <Label lang="en">off</Label>
+            </Item>
+        </Parameter>
+    </Configuration>
+
+    <Associations>
+        <Group>
+            <Index>1</Index>
```
Comment (cdjackson) [inline bundles/binding/org.openhab.binding.zwave/database/dragontech/pa100.xml]: Is this group one that you would expect to be linked to openhab? If you want it to send reports to the binding, you need to add ```<SetToController>true</SetToController>``` in the group.

File: bundles/binding/org.openhab.binding.zwave/database/dragontech/wd100.xml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+            <Help lang="en">The number of tens of milliseconds (1-255) to delay on each dimming step.</Help>
+        </Parameter>
+    </Configuration>
+
+    <Associations>
+        <Group>
+            <Index>1</Index>
+            <Maximum>5</Maximum>
+            <Label lang="en">Group 1</Label>
+        </Group>
```
Comment (cdjackson) [inline bundles/binding/org.openhab.binding.zwave/database/dragontech/wd100.xml]: Is this group one that you would expect to be linked to openhab? If you want it to send reports to the binding, you need to add <SetToController>true</SetToController> in the group.

File: bundles/binding/org.openhab.binding.zwave/database/dragontech/ws100.xml [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+            <Item>
+                <Value>1</Value>
+                <Label lang="en">Inverted</Label>
+            </Item>
+        </Parameter>
+    </Configuration>
+
+    <Associations>
+        <Group>
+            <Index>1</Index>
```
Comment (cdjackson) [inline bundles/binding/org.openhab.binding.zwave/database/dragontech/ws100.xml]: Is this group one that you would expect to be linked to openhab? If you want it to send reports to the binding, you need to add <SetToController>true</SetToController> in the group.

File: bundles/binding/org.openhab.binding.zwave/database/schlage/be468.xml [hunk 4]
```diff
@@ -0,0 +1,109 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Product>
+    <Model>PA-100</Model>
```
Comment (cdjackson) [inline bundles/binding/org.openhab.binding.zwave/database/schlage/be468.xml]: I guess this is a cut and paste error?

File: bundles/binding/org.openhab.binding.zwave/database/schlage/be468.xml [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+    <Label lang="en">Plug-in On/Off Switch</Label>
```
Comment (cdjackson) [inline bundles/binding/org.openhab.binding.zwave/database/schlage/be468.xml]: Again - cut and paste error

File: bundles/binding/org.openhab.binding.zwave/database/schlage/be468.xml [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+            <Default>4</Default>
+            <Size>1</Size>
+            <Minimum>4</Minimum>
+            <Maximum>8</Maximum>
+            <Label lange="en">User Code PIN Length</Label>
+            <Help lang="en">User codes can be 4-8 digits. Changing length will delete all existing codes.</Help>
+        </Parameter>
+    </Configuration>
+
+    <Associations>
```
Comment (cdjackson) [inline bundles/binding/org.openhab.binding.zwave/database/schlage/be468.xml]: I don't know what the association groups do, but if one (or both) of them are meant to report information to a controller then you should set ```<SetToController>true</SetToController>``` in the group.

**Comment by cdjackson** [comment]
Thanks - just a couple of small comments in the database files....

File: bundles/binding/org.openhab.binding.zwave/database/dragontech/wd100.xml (hunk 2 above)
Comment (llamahunter) [inline bundles/binding/org.openhab.binding.zwave/database/dragontech/wd100.xml]: Um, yes, but I'm not sure what you mean by "add true"? 

Richard

**Comment by llamahunter** [comment]
Yes, thanks for catching the cut and paste errors in the be468 file.  I was not able to really test that one since almost all communication seems to require the security command class, which is not yet implemented.

There's no clear documentation of the XML schema for these files that I could find.  I was sort of flying blind and did not know about the SetToController attribute.

**Comment by cdjackson** [comment]
Doc for the XML is here - it’s reasonably up to date I think -:
https://github.com/cdjackson/HABmin2/wiki/Z-Wave-Product-Database <https://github.com/cdjackson/HABmin2/wiki/Z-Wave-Product-Database>

**Comment by llamahunter** [comment]
It looks like very few other switches have SetToController set correctly.  None of the Linear, Leviton, or GE switches do.

**Comment by cdjackson** [comment]
> It looks like very few other switches have SetToController set correctly. None of the Linear, Leviton, or GE switches do.
> 
I know, but that doesn’t make it right :)

This was something that I had in the database from the beginning, but as it wasn’t used until reasonably recently, and also as I wasn’t reviewing the files until late last year (Kai and Thomas were) this generally wasn’t added. I’m now trying to ensure this is added in any new files as it makes it much easier for new users.

Feel free to add this to any switches you have if you know if should be set - it would be much appreciated.

**Comment by llamahunter** [comment]
Would this explain why local changes to the switch weren't getting timely updates in openhab?  Was going to file a bug on that.

**Comment by cdjackson** [comment]
Probably. Associations are the method that is normally used to advise of state changes made directly to the switch. There are others, but this is the main one, and if it’s not set, then you don’t get updates.

**Comment by llamahunter** [comment]
One more question... the datasheet for a lock very similar to mine says it has only has group 1 with two members, but openzwave-control-panel says that, after talking to the lock, it has two groups, group 1 with max one member, and group 2 with max three members.  Which should I put into the database?

**Comment by cdjackson** [comment]
Yes - put them both in and then do a test to see if it's reported. From HABmin, you can click the REFRESH button to ask for an update of the group - then check the debug logs to see if it worked or if it timed out.

**Comment by llamahunter** [comment]
Unfortunately, I can't test the associations or configurations for the schlage lock.  The zwave library says it has none (thus the reason for the null check I added).  I think this is because the lock requires all communication to be done using the security command class, which openhab does not yet support.  So, I'm going to go with what's in the datasheet.

**Comment by llamahunter** [comment]
hmm... I might have messed this up.  I don't use git very much at work.  Some unexpected stuff ended up in the pull request.

**Comment by cdjackson** [comment]
Yes - there seems to be some stuff in there that will stop me from merging - can you maybe cherry pick the good bits into a separate PR?

**Comment by llamahunter** [comment]
Split into three separate pull requests

**Pull request closed without merging**